src/Markup/Avalonia.Markup.Xaml/XamlIl/CompilerExtensions/AvaloniaXamlIlLanguage.cs

```diff
@@ -35,6 +35,7 @@ public static XamlIlLanguageTypeMappings Configure(IXamlIlTypeSystem typeSystem)
                 },
                 ProvideValueTarget = typeSystem.GetType("Avalonia.Markup.Xaml.IProvideValueTarget"),
                 RootObjectProvider = typeSystem.GetType("Avalonia.Markup.Xaml.IRootObjectProvider"),
+                RootObjectProviderIntermediateRootPropertyName = "IntermediateRootObject",
```

Comment:
Do we use the `IntermediateRootObject` concept anywhere after switching to a NameScope stack?

Reply:
I think it might be useful in general by markup extensions and converters. It doesn't have a performance impact, so I'd like to keep it.